shotover/src/codec/cassandra.rs

```diff
@@ -768,15 +878,11 @@ impl CassandraEncoder {
         Ok((uncompressed_len, compressed_len))
     }
 
-    fn encode_envelope(
-        &mut self,
-        dst: &mut BytesMut,
-        m: Message,
-        envelope_compresson: Compression,
-    ) -> Result<()> {
+    fn encode_envelope(&mut self, m: Message, envelope_compresson: Compression) -> Result<Bytes> {
         // TODO: always check if cassandra message
-        match m.into_encodable(MessageType::Cassandra)? {
+        Ok(match m.into_encodable(MessageType::Cassandra)? {
             Encodable::Bytes(bytes) => {
+                // TODO this is a weird side effect to keep in this function
```

Comment:
Going to handle this in a followup PR, but this method has some weird side effects and needs to be called in a magic order for the codec to work that isn't clear at a first glance.